httpcore/_async/connection_pool.py

```diff
@@ -299,11 +299,16 @@ def _assign_requests_to_connections(self) -> List[AsyncConnectionInterface]:
         return closing_connections
 
     async def _close_connections(self, closing: List[AsyncConnectionInterface]) -> None:
+        if not closing:
+            return
```

Comment:
Could you explain this change to me?

Reply:
If its an empty list then we dont need to go through `await async_cancel_shield` below, right? If that was the question